### PR TITLE
Add new sample cubemap node with world space direction

### PR DESCRIPTION
**Pull request by sebastienlagarde**
### Purpose of this PR
Added a new **Sample Cubemap** Node which take the world space direction to sample it.
(High demanding change from artists)
Rename the old Sample Cubemap to **Sample Reflected Cubemap** to highlight that it is performing a reflection along from the view direction and object normal. Functionnality is unchanged.

image

image

image

---
### Testing status
I have added a new input node test in existing test of shader graph project
I have tested the new node with world space normal and a provided vector3 and a sky cubemap. All this to works well.

Note: I am not sure about what is a cubemap asset, maybe there is something to check here? But my guess is that it is independent of this change.

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Comment by sebastienlagarde** [comment]
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/sg%252Fadd-sample-cubemap

File: com.unity.shadergraph/Documentation~/Sample-Reflected-Cubemap-Node.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Sample Cubemap Node
```
Comment (alindmanUnity) [inline com.unity.shadergraph/Documentation~/Sample-Reflected-Cubemap-Node.md:1]: should be Sample Reflected Cubemap

File: com.unity.shadergraph/Documentation~/Sample-Reflected-Cubemap-Node.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+## Description
+
+Samples a **Cubemap** with reflected vector and returns a **Vector 4** color value for use in the shader. Requires **View Direction** and **Normal** inputs to sample the **Cubemap**. You can achieve a blurring effect by sampling at a different Level of Detail using the **LOD** input. You can also define a custom **Sampler State** using the **Sampler** input.
+
```
Comment (alindmanUnity) [inline com.unity.shadergraph/Documentation~/Sample-Reflected-Cubemap-Node.md:6]: probably worth noting that it was previously Sample Cubemap?

**Review by alindmanUnity — APPROVED** [review]
Did you update the result images for the input node tests? I don't see any png file changes

File: com.unity.shadergraph/Documentation~/Sample-Reflected-Cubemap-Node.md (hunk 2 above)
Comment (sebastienlagarde) [reply com.unity.shadergraph/Documentation~/Sample-Reflected-Cubemap-Node.md:6]: I will let the reply to @sharlenet . I don't know. I think this is the purpose of the upgrade guide to say that?
I think the doc should be what is the current state of node. But open to change if needed.

**Comment by sebastienlagarde** [comment]
> Did you update the result images for the input node tests? I don't see any png file changes

Nope, waiting for Yamato to generate all screenshots on all platforms

**Review by cdxntchou — APPROVED** [review]

File: com.unity.shadergraph/Documentation~/Upgrade-Guide-10-0-x.md [hunk 3]
```diff
@@ -4,6 +4,10 @@
 
 Shader Graph has renamed the **Vector 1** property as **Float** in both the Vector 1 node and the exposed parameter list. The **Float** precision was also renamed as **Single**. Behavior is exactly the same, and only the names have changed.
 
+## Renamed Sample Cubemap
+
+Shader Graph has renamed the previous Sample Cubemap Node in Sample Reflected Cubemap and have added a new Sample Cubemap which used world space direction.
```
Comment (LandonTownsendUnity) [inline com.unity.shadergraph/Documentation~/Upgrade-Guide-10-0-x.md:9]: little grammar fix here, 
"Shader Graph has renamed the previous Sample Cubemap Node **to** Sample Reflected Cubemap**,** and **has** added a new Sample Cubemap**,** which **uses** world space direction."

File: com.unity.shadergraph/Documentation~/Sample-Cubemap-Node.md [hunk 4]
```diff
@@ -2,15 +2,14 @@
 
 ## Description
 
-Samples a **Cubemap** and returns a **Vector 4** color value for use in the shader. Requires **View Direction** and **Normal** inputs to sample the **Cubemap**. You can achieve a blurring effect by sampling at a different Level of Detail using the **LOD** input. You can also define a custom **Sampler State** using the **Sampler** input.
+Samples a **Cubemap** and returns a **Vector 4** color value for use in the shader. Requires **Direction** inputs in world space to sample the **Cubemap**. You can achieve a blurring effect by sampling at a different Level of Detail using the **LOD** input. You can also define a custom **Sampler State** using the **Sampler** input.
```
Comment (LandonTownsendUnity) [inline com.unity.shadergraph/Documentation~/Sample-Cubemap-Node.md:5]: "inputs" should be "input" since there's only one

**Review by ghost — CHANGES_REQUESTED** [review]
A few minor grammar fixes

**Review by ghost — APPROVED** [review]
So happy to see this!
image
Only tests I think this needed were 1. basic use case and 2. upgrade test with shadergraph containing old Sample Cubemap node (which passed). Aside from the minor grammar fixes, this is good to go!

File: com.unity.shadergraph/Documentation~/Input-Nodes.md [hunk 5]
```diff
@@ -90,7 +90,7 @@
 
 ## Texture
 
-|[Cubemap Asset](Cubemap-Asset-Node.md)|[Sample Cubemap](Sample-Cubemap-Node.md)|
+|[Cubemap Asset](Cubemap-Asset-Node.md)|[Sample Cubemap](Sample-Cubemap-Node.md)|[Sample Cubemap](Sample-Reflected-Cubemap-Node.md)|
```
Comment (sharlenet) [inline com.unity.shadergraph/Documentation~/Input-Nodes.md:93]: I don't understand why a third column was added here. Should it just replace the existing `[Sample Cubemap](Sample-Cubemap-Node.md)`?

File: com.unity.shadergraph/Documentation~/Sample-Cubemap-Node.md (hunk 4 above)
Comment (sharlenet) [reply com.unity.shadergraph/Documentation~/Sample-Cubemap-Node.md:5]: Thanks for pointing this out! I'll fix it.

File: com.unity.shadergraph/Documentation~/Input-Nodes.md (hunk 5 above)
Comment (sebastienlagarde) [reply com.unity.shadergraph/Documentation~/Input-Nodes.md:93]: I just don't know what this stuff was :), will revert

File: com.unity.shadergraph/Documentation~/Upgrade-Guide-10-0-x.md (hunk 3 above)
Comment (sebastienlagarde) [reply com.unity.shadergraph/Documentation~/Upgrade-Guide-10-0-x.md:9]: thanks fixed

**Comment by sebastienlagarde** [comment]
Yamato is green

**Pull request merged**